### ISSUE TITLE
Struct skeleton with pointer to element in a slice not found

### ISSUE THREAD

**Issue by ThisAccountHasBeenSuspended** [zigtools/zls#1872]
### Zig Version

0.12.0

### Zig Language Server Version

07508440588bf06f041da9f6dd3956948368f194

### Client / Code Editor / Extensions

VSCode `Zig Language`

### Steps to Reproduce and Observed Behavior

```zig
const std = @import("std");

const Blub = struct {
    pub fn printHello(_: @This()) void {
        std.debug.print("Hello\n", .{});
    }
};

pub fn main() anyerror!void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    const allocator = gpa.allocator();
    defer _ = gpa.deinit();

    var array = std.ArrayList(?Blub).init(allocator);
    defer array.deinit();
    try array.append(Blub{});

    for (array.items[0..]) |*item| {
        // Function not found
        item.*.?.printHello();
    }

    for (array.items[0..]) |item| {
        // Function found
        item.?.printHello();
    }
}
```

![image](https://github.com/zigtools/zls/assets/154023155/b7a1de58-480b-4fff-85ea-2fdbbbacdd6d)

### Expected Behavior

Get the same result as with a "normal" variable or without a pointer.

(Image created with GIMP)
![image](https://github.com/zigtools/zls/assets/154023155/363d87d3-fb12-4647-a11c-cd106e9f34b4)

**Comment by ThisAccountHasBeenSuspended** [comment]
> Hmm

Tested and now it works in my projects and this example above.

![image](https://github.com/zigtools/zls/assets/154023155/51bc5262-4e5d-41be-88d0-68072273f891)

**Comment by wizzymore** [comment]
Can confirm, i was following the ziglings exercises and had a little bit of trouble at exercise 58 because of this:
<img width="563" alt="image" src="https://github.com/zigtools/zls/assets/25827787/6fd37480-31da-4519-9407-e07fd687b9df">
Instead of `*?NotebookEntry` i get `?NotebookEntry` and when trying to return the correct `&entry.*.?` i get no autocomplete. I get the autocomplete for entry.?.something which is wrong.

**Comment by Techatrix** [comment]
fixed by #1893
